authentication/models.py

```diff
@@ -0,0 +1,78 @@
+"""User Model
+
+This model is used to manage the user profile. This model is also used to
+create table in database by Django. This model inherits from the AbstractUser.
+This is extend-user model. Email is used as username field. Username will be
+none in this model.
+"""
+
+
+from django.db import models
+from django.contrib.auth.models import AbstractUser, BaseUserManager
+from django.utils.translation import gettext_lazy as _
+
+
+class UserManager(BaseUserManager):
+    """Define a model manager for User model with no username field."""
+
+    use_in_migrations = True
+
+    def _create_user(self, email, password, **extra_fields):
+        """Create and save a User with the given email and password."""
+
+        if not email:
+            raise ValueError("The given email must be set")
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    # def create_user(self, email, password=None, **extra_fields):
```

Comment:
I wonder why we have commented out this `create_user` overriding? 
as the default, `create_user` function uses the username.